packages/expo-mail-composer/ios/MailComposerExceptions.swift

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+import MessageUI
+import ExpoModulesCore
+
+internal final class CannotSendMailException: Exception {
+  override var reason: String {
+    "Mail services are not available, make sure you're signed into the Mail app"
+  }
+}
+
+internal final class UnfinishedSessionException: Exception {
+  override var reason: String {
+    "Cannot compose an email while an unfinished session already exists"
+  }
+}
```

Comment:
I would not leak `session` word to the user space as it might be confusing. From the user's perspective it's just an `async function` that could not be invoked concurrently multiple times.
```suggestion
internal final class AnotherOperationInProgressException: Exception {
  override var reason: String {
    "Cannot compose an email because another email composing operation is in progress"
  }
}
```